PostHog/PostHogApi.swift

```diff
@@ -217,4 +237,45 @@ class PostHogApi {
             }
         }.resume()
     }
+
+    func remoteConfig(
+        completion: @escaping ([String: Any]?, _ error: Error?) -> Void
+    ) {
+        guard let request = getRemoteConfigRequest() else {
+            hedgeLog("Error calling the remote config API: unable to create request")
+            return
+        }
+
+        let config = sessionConfig()
+
+        let task = URLSession(configuration: config).dataTask(with: request) { data, response, error in
+            if let error {
+                hedgeLog("Error calling the remote config API: \(error.localizedDescription)")
+                return completion(nil, error)
+            }
+
+            let httpResponse = response as! HTTPURLResponse
+
+            if !(200 ... 299 ~= httpResponse.statusCode) {
```

Comment:
we can probably double check with ben that is the expected errors here and if there's any special case.